spec/durable-promise.yaml

```diff
@@ -1,350 +1,217 @@
-                tags: 
-                  type: object
-                  additionalProperties:
-                    type: string
-        Value:
-            type: object
-            properties:
-                data:
-                    type: string
-                headers:
-                    type: object
-                    additionalProperties:
-                        type: string
-        CreatePromiseRequest:
-            type: object
-            properties:
-                id:
-                    type: string
-                param:
-                    $ref: "#/components/schemas/Value"
-                timeout:
-                    type: integer
-        ResolvePromiseRequest:
-            type: object
-            properties:
-                value:
-                    $ref: "#/components/schemas/Value"
-        RejectPromiseRequest:
-            type: object
-            properties:
-                value:
-                    $ref: "#/components/schemas/Value"
-        CancelPromiseRequest:
-            type: object
-            properties:
-                value:
-                    $ref: "#/components/schemas/Value"
-        SearchPromiseResponse: 
-            type: object 
-            properties: 
-              cursor: 
-                type: string 
-              promises: 
-                type: array
-                items:
-                  $ref: "#/components/schemas/Promise"
-              
+  parameters:
+    QueryFilters:
+      name: filters
+      in: query
+      style: form # Use form style to group params
+      explode: true
+      schema:
+        type: object
+        properties:
+          id:
+            type: string
+            description: |
+              Fuzzy search query string to find promises via fuzzy IDs. This string will be matched flexibly 
+              against IDs in the system, so it does not need to exactly match the full ID.
+
+              Some examples of valid fuzzy ID strings:
+              - "abc123*" - Would match ID "abc123def" 
+              - "pro*fil" - Would match IDs containing "pro123file"
+          state:
+            $ref: "#/components/schemas/PromiseState"
+          limit:
+            type: integer
+            description: Number of results
+          cursor:
+            type: string
+            description: Cursor for pagination
+
+    PathId:
+      name: id
+      in: path
+      required: true
+      description: The promise ID
+      schema:
+        type: string
+
+    IdempotencyKey:
+      name: idempotency-key
+      in: header
+      description: Deduplicates multiple requests
+      schema:
+        type: string
+    StrictMode:
+      name: strict
+      in: header
+      description: If true, deduplicates only when promise is pending
+      schema:
+        type: boolean
+    RequestId:
+      name: request-id
+      in: header
+      description: Unique ID for each request
+      schema:
+        type: string
+
+  requestBodies:
```

Comment:
Could we move CreatePromise and CompletePromise up to top level objects and remove this?